node/node.go

```diff
@@ -129,7 +130,8 @@ type Node struct {
 	err              error
 	agent            *agent.Agent
 	manager          *manager.Manager
-	notifyNodeChange chan *api.Node // used to send role updates from the dispatcher api on promotion/demotion
+	notifyNodeChange chan *agent.NodeChanges // used to send role updates from the dispatcher api on promotion/demotion
```

Comment:
Are we doing this (having node changes and TLSChanges separate) so that nodes don't rotate their certificates twice for a root rotation?